src/index.py

```diff
@@ -79,3 +96,260 @@ def contents_from_page(page: str) -> str:
     """
     contents = page.split(NAVIGATION_TABLE_START)
     return contents[0]
+
+
+class _ParsedListItem(typing.NamedTuple):
+    """Represents a parsed item in the contents table.
+
+    Attrs:
+        whitespace_count: The number of leading whitespace characters
+        reference_title: The name of the reference
+        reference_value: The link to the referenced item
+        rank: The number of preceding elements in the list
+    """
+
+    whitespace_count: int
+    reference_title: str
+    reference_value: str
+    rank: int
+
+
+def _parse_item_from_line(line: str, rank: int) -> _ParsedListItem:
```

Comment:
I suppose this is intended but this will supports things like:
```
- [line1](line1)
  - [line1.1](line1.1)
       - [line1.1.1](line1.1.1)
- [line2](line2)
       - [line2.1](line2.1)
 ````
With line2.1 being hierarchy 2 and line1.1.1 being hierarchy 3

Reply:
Closing this comment, we can continue the discussion in the thread where I've linked a commit to handle this